File: backend/bin/src/services/stripe.service.js
const Stripe = require('stripe');
require('dotenv').config();

const stripeSecret = process.env.STRIPE_SECRET_KEY;
const isMockStripe = !stripeSecret || stripeSecret.includes('mockkey');

let stripeInstance;
if (!isMockStripe) {
  stripeInstance = new Stripe(stripeSecret, {
    apiVersion: '2022-11-15'
  });
} else {
  console.log('Stripe API Key is missing or using a mock token. Payment flows will run in Sandbox Mock mode.');
}

const createPaymentIntent = async (amountInCents, metadata = {}) => {
  if (isMockStripe) {
    console.log(`[MOCK STRIPE] Creating Payment Intent for $${(amountInCents / 100).toFixed(2)}`);
    return {
      id: `pi_mock_${Math.random().toString(36).substring(2, 15)}`,
      client_secret: `pi_mock_secret_${Math.random().toString(36).substring(2, 15)}`,
      amount: amountInCents,
      status: 'requires_payment_method'
    };
  } else {
    try {
      const paymentIntent = await stripeInstance.paymentIntents.create({
        amount: amountInCents,
        currency: 'usd',
        metadata
      });
      return paymentIntent;
    } catch (error) {
      console.error('Stripe PaymentIntent creation failure:', error);
      throw error;
    }
  }
};

const refundPayment = async (paymentIntentId) => {
  if (isMockStripe) {
    console.log(`[MOCK STRIPE] Refunding Payment Intent: ${paymentIntentId}`);
    return {
      id: `re_mock_${Math.random().toString(36).substring(2, 15)}`,
      status: 'succeeded'
    };
  } else {
    try {
      const refund = await stripeInstance.refunds.create({
        payment_intent: paymentIntentId
      });
      return refund;
    } catch (error) {
      console.error('Stripe refund failure:', error);
      throw error;
    }
  }
};

module.exports = {
  createPaymentIntent,
  refundPayment
};

File: backend/src/main/java/com/telehealth/pro/service/StripeService.java
package com.telehealth.pro.service;

import com.stripe.Stripe;
import com.stripe.model.PaymentIntent;
import com.stripe.model.Refund;
import com.stripe.param.PaymentIntentCreateParams;
import com.stripe.param.RefundCreateParams;
import jakarta.annotation.PostConstruct;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.util.Map;
import java.util.UUID;

@Service
public class StripeService {

    @Value("${stripe.api.key}")
    private String stripeSecretKey;

    private boolean isMockStripe;

    @PostConstruct
    public void init() {
        isMockStripe = stripeSecretKey == null || stripeSecretKey.isEmpty() || stripeSecretKey.contains("mockkey");
        if (!isMockStripe) {
            Stripe.apiKey = stripeSecretKey;
            System.out.println("[STRIPE] Initialized with real Stripe secret key.");
        } else {
            System.out.println("Stripe API Key is missing or using a mock token. Payment flows will run in Sandbox Mock mode.");
        }
    }

    public StripeIntentResult createPaymentIntent(long amountInCents, Map<String, String> metadata) {
        if (isMockStripe) {
            String mockId = "pi_mock_" + UUID.randomUUID().toString().replace("-", "").substring(0, 12);
            String mockSecret = "pi_mock_secret_" + UUID.randomUUID().toString().replace("-", "").substring(0, 16);
            System.out.println("[MOCK STRIPE] Creating Payment Intent for $" + (amountInCents / 100.0));
            return StripeIntentResult.builder()
                    .id(mockId)
                    .clientSecret(mockSecret)
                    .amount(amountInCents)
                    .status("requires_payment_method")
                    .build();
        } else {
            try {
                PaymentIntentCreateParams params = PaymentIntentCreateParams.builder()
                        .setAmount(amountInCents)
                        .setCurrency("usd")
                        .putAllMetadata(metadata)
                        .build();
                PaymentIntent intent = PaymentIntent.create(params);
                return StripeIntentResult.builder()
                        .id(intent.getId())
                        .clientSecret(intent.getClientSecret())
                        .amount(intent.getAmount())
                        .status(intent.getStatus())
                        .build();
            } catch (Exception e) {
                System.err.println("[STRIPE ERROR] PaymentIntent creation failure: " + e.getMessage());
                throw new RuntimeException("Stripe error: " + e.getMessage(), e);
            }
        }
    }

    public void refundPayment(String paymentIntentId) {
        if (isMockStripe) {
            System.out.println("[MOCK STRIPE] Refunding Payment Intent: " + paymentIntentId);
            return;
        } else {
            try {
                RefundCreateParams params = RefundCreateParams.builder()
                        .setPaymentIntent(paymentIntentId)
                        .build();
                Refund.create(params);
                System.out.println("[STRIPE] Refund processed successfully: " + paymentIntentId);
            } catch (Exception e) {
                System.err.println("[STRIPE ERROR] Refund failure: " + e.getMessage());
                throw new RuntimeException("Stripe refund error: " + e.getMessage(), e);
            }
        }
    }
}
